pkg/objcache/objcache.go

```diff
@@ -146,35 +169,74 @@ func (c *Cache) Create(key string, size int64) (w io.WriteCloser, err error) {
 // Open - open the in-memory file, returns an in memory read seeker.
 // returns an error ErrNotFoundInCache, if the key does not exist.
 func (c *Cache) Open(key string) (io.ReadSeeker, error) {
-	c.mutex.RLock()
-	defer c.mutex.RUnlock()
-
 	// Entry exists, return the readable buffer.
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
 	buf, ok := c.entries[key]
 	if !ok {
 		return nil, ErrKeyNotFoundInCache
 	}
+	buf.lastAccessed = time.Now().UTC()
 	return bytes.NewReader(buf.value), nil
 }
 
-// Delete - delete deletes an entry from in-memory fs.
+// Delete - delete deletes an entry from the cache.
 func (c *Cache) Delete(key string) {
 	c.mutex.Lock()
-	defer c.mutex.Unlock()
+	c.delete(key)
+	c.mutex.Unlock()
+	if c.OnEviction != nil {
+		c.OnEviction(key)
+	}
+}
 
-	// Delete an entry.
-	buf, ok := c.entries[key]
-	if ok {
-		c.deleteEntry(key, len(buf.value))
+// DeleteExpired - deletes all the expired entries from the cache.
+func (c *Cache) DeleteExpired() {
+	var evictedEntries []string
+	c.mutex.Lock()
+	for k, v := range c.entries {
+		if v.isExpired(c.expiration) {
+			c.delete(k)
+			evictedEntries = append(evictedEntries, k)
+		}
+	}
+	c.mutex.Unlock()
+	for _, k := range evictedEntries {
+		if c.OnEviction != nil {
+			c.OnEviction(k)
+		}
 	}
 }
 
-// Deletes the entry that was found.
-func (c *Cache) deleteEntry(key string, size int) {
-	delete(c.entries, key)
-	c.currentSize -= uint64(size)
-	c.totalEvicted++
-	if c.OnEviction != nil {
-		c.OnEviction(key)
+// StopJanitor sends a message to the janitor routine to stop.
+func (c *Cache) StopJanitor() {
```

Comment:
can be `Stop` or `Destroy` as `janitor` term need not be exposed to the cache consumer